### DISCUSSION TITLE
Change log for IP addresses contained in a particular prefix/range

### DISCUSSION THREAD

**Discussion by bluikko** [netbox-community/netbox#10855] [Ideas]
I was trying to look for what are the latest changes to IP addresses contained in a particular prefix. The obvious way to attempt this was to browse to the prefix and check the "Change log" page. But this page contains log of changes only to the prefix itself and does not include any changes to the IP addresses contained in the prefix. Next I checked change logs of the child ranges contained in the prefix but the situation is the same.

Is there a way to get list of changes to IP addresses contained in a particular prefix or a range contained therein?  
There seems to be no way at least in the web interface or then I just cannot find it.  
Also I could not find any issues or discussions about this feature. Am I the only one looking for such a feature? Would implementing such a feature be not possible due to performance reasons or such?

**Comment by mtinberg** [comment]
I looked at the API Docs for /extras/changelog to see what filtering options exist (I think all API options work in the web UI but many are not listed in the filtering web forms)

/extras/changelog/?changed_object_type=ipam.ipaddress&object_repr__isw=192.0.2.

This is a string match on the beginning of the object name, so it's not CIDR aware, but this can probably get you close to what you want.  If you want to be more exact, then you'll need to look up the ID number for each IP Address record you want and then pass a bunch of changed_object_id parameters (limited by the total URL length configured in your web app server), but this could also miss deleted records (because they don't have active record IDs) that the name based match would show.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: bluikko ***@***.***>
Sent: Saturday, November 5, 2022 3:44 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Change log for IP addresses contained in a particular prefix/range (Discussion #10855)


I was trying to look for what are the latest changes to IP addresses contained in a particular prefix. The obvious way to attempt this was to browse to the prefix and check the "Change log" page. But this page contains log of changes only to the prefix itself and does not include any changes to the IP addresses contained in the prefix. Next I checked change logs of the child ranges contained in the prefix but the situation is the same.

Is there a way to get list of changes to IP addresses contained in a particular prefix or a range contained therein?
There seems to be no way at least in the web interface or then I just cannot find it.
Also I could not find any issues or discussions about this feature. Am I the only one looking for such a feature? Would implementing such a feature be not possible due to performance reasons or such?
